src/snowflake/cli/plugins/object/stage/manager.py

```diff
@@ -164,3 +186,90 @@ def create(self, stage_name: str, comment: Optional[str] = None) -> SnowflakeCur
     def iter_stage(self, stage_path: str):
         for file in self.list_files(stage_path).fetchall():
             yield file["name"]
+
+    def execute(
+        self,
+        stage_path: str,
+        on_error: OnErrorType,
+        variables: Optional[List[str]] = None,
+    ):
+        all_files_list = self._get_files_list_from_stage(stage_path)
+        # filter files from stage if match stage_path pattern
+        filtered_file_list = self._filter_files_list(stage_path, all_files_list)
+
+        if not filtered_file_list:
+            raise ClickException(f"No files matched pattern '{stage_path}'")
+
+        # sort filtered files in alphabetical order with directories at the end
+        sorted_file_list = sorted(
+            filtered_file_list, key=lambda f: (path.dirname(f), path.basename(f))
+        )
+
+        sql_variables = self._parse_execute_variables(variables)
+        results = []
+        for file in sorted_file_list:
+            results.append(self._call_execute_immediate(file, sql_variables, on_error))
+
+        return results
+
+    def _get_files_list_from_stage(self, stage_path: str) -> List[str]:
+        stage_name = self.get_stage_name_from_path(stage_path)
+        files_list_result = self.list_files(stage_name).fetchall()
+
+        if len(files_list_result) == 0:
+            raise ClickException(f"No files found on stage '{stage_name}'")
+
+        return [f["name"] for f in files_list_result]
+
+    def _filter_files_list(
+        self, stage_path: str, files_on_stage: List[str]
+    ) -> List[str]:
+        stage_path = self.remove_stage_prefix(stage_path)
+
+        # Execute everything if only stage name was provided
+        if "/" not in stage_path:
+            return files_on_stage
+
+        # Filter with fnmatch if contains `*` or `?`
+        if glob.has_magic(stage_path):
+            return fnmatch.filter(files_on_stage, stage_path)
+
+        # Exact file path was provided if stage_path in file list
+        if stage_path in files_on_stage:
```

Comment:
How about not existing file?  If the file does not exist then the later comment is not true (directory was not privded).

How about `if Path(stage_path).suffix in SUPPORTED_FILES_FORMATS` where `SUPPORTED_FILES_FORMATS = {".sql"}`. WDYT?

Reply:
It depends if you provide stage that does not exist, command raises exception from Snowflake. If you provide file does not exist on existing stage message is `No files matched pattern 'exe/*.txt'`. 

I forgot to add extension check.